test/git-open.bats

```diff
@@ -269,6 +269,35 @@ setup() {
 
 }
 
+
+@test "bitbucket server with different root context" {
+  # https://github.com/paulirish/git-open/pull/15
+  git remote set-url origin "https://user@bitbucket.example.com/git/scm/ppp/test-repo.git"
+  run ../git-open
+  assert_output "https://bitbucket.example.com/git/projects/ppp/repos/test-repo"
+}
+
+
+@test "bitbucket server with different root context with multiple parts" {
+  # https://github.com/paulirish/git-open/pull/15
+  git remote set-url origin "https://user@bitbucket.example.com/really/long/root/context/scm/ppp/test-repo.git"
+  run ../git-open
+  assert_output "https://bitbucket.example.com/really/long/root/context/projects/ppp/repos/test-repo"
+}
+
+
+@test "bitbucket: Bitbucket Server private user repos with different root context" {
+  # https://github.com/paulirish/git-open/pull/83#issuecomment-309968538
+  git remote set-url origin "https://mybb.domain.com/root/context/scm/~first.last/rrr.git"
+  git checkout -B "develop"
+  run ../git-open
+  assert_output "https://mybb.domain.com/root/context/projects/~first.last/repos/rrr/browse?at=develop" ||
+    assert_output "https://mybb.domain.com/root/context/projects/~first.last/repos/rrr/browse?at=refs%2Fheads%2Fdevelop" ||
+    assert_output "https://mybb.domain.com/root/context/projects/~first.last/repos/rrr/browse?at=refs/heads/develop"
+
```

Comment:
#nitpick - remove the space?

Reply:
Saw this too, but it's indentation for the chained `||`.

We could possibly reduce duplicate strings by making use of `assert_output --regexp`, although it would be less readable. Like so:

```bash
assert_output --regexp "https://mybb.domain.com/root/context/projects/~first.last/repos/rrr/browse\?at=(refs(/|%2F)heads(/|%2F))?develop"
```

Reply:
I've simply copied this part from the other existing unit tests, including the `||` chaining and the indentation.

Not a huge fan of the regexp check, due to readability...